ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -918,3 +924,20 @@ func (ipfs *Connector) SwarmPeers() (api.SwarmPeers, error) {
 	}
 	return swarm, nil
 }
+
+// extractCid extracts the cid argument from a url.URL, either via
+// the query string parameters or from the url path itself.
+func extractCid(u *url.URL) (string, bool) {
+	arg := u.Query().Get("arg")
+	if arg != "" {
+		return arg, true
+	}
+
+	p := strings.TrimPrefix(u.Path, "/api/v0/")
+	segs := strings.Split(p, "/")
+
+	if len(segs) > 2 {
+		return segs[len(segs)-1], true
+	}
```

Comment:
Maybe I'm starting to get what you were saying yesterday..

In general, we have things like:

`pin/ls/<arg>`

`cat/<arg>`

but from here we have no way of distinguishing which commands are valid in order to know what part of the url is the argument.

Let's print a warning "You are using an undocumented form of the IPFS API. Consider passing your command arguments with the '?arg=<argument>' query parameter".

For the moment it works like this I guess...

Reply:
We'll we do know which subcommands we are handling, so that's our list if we want to be strict.

Reply:
> Let's print a warning "You are using an undocumented form of the IPFS API. Consider passing your command arguments with the '?arg=' query parameter".

@hsanjuan should I be returning that in the json response, as it is the client that needs to change? 

Also, the way that we have the IPFS-Cluster api documented in the README suggests that `pin/ls/<arg>` is the more appropriate way to interact with the api, so that should probably get changed as well.

Reply:
> @hsanjuan should I be returning that in the json response, as it is the client that needs to change?

No, just spam the logs

> `/pin/ls/<arg>` that refers to the cluster API. This is different from the IPFS API.